source/test/F0.Analyzers.Tests/Assertions/AssertionExtensions.cs

```diff
@@ -0,0 +1,14 @@
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace F0.Tests.Assertions;
+
+internal static class AssertionExtensions
+{
+	internal static void ShouldNotBeNull<T>([NotNull] this T reference, string because = "", params object[] becauseArgs)
+		where T : class?
```

Comment:
@NKnusperer your suggestion was:
```CSharp
internal static class Flash0verAssert
{
#pragma warning disable CS8777 // Parameter must have a non-null value when exiting.
	internal static void NotNull([System.Diagnostics.CodeAnalysis.NotNull] this object? instance)
	{
		instance.Should().NotBeNull();

	}
#pragma warning restore CS8777 // Parameter must have a non-null value when exiting.
}
```

I named it `ShouldNotBeNull` to reveal that it's semantically equivalent with `.Should().NotBeNull()`, and also adopted _Fluent Assertions_ parameters.

What do you think?

Reply:
Personally I would still preferer the suppression via `#pragma` because why would you check the same thing twice?
It's your call.

Reply:
You're right. Wasted cycles when running the tests in a _Debug_ build.
I'll `pragma`ify it.